sos/plugins/kubernetes.py

```diff
@@ -60,70 +50,73 @@ def setup(self):
-
-            # get all namespaces in use
-            kn = self.get_command_output('%s get namespaces' % kube_cmd)
-            knsps = [n.split()[0] for n in kn['output'].splitlines()[1:] if n]
-
-            resources = [
-                'limitrange',
-                'pods',
-                'pvc',
-                'rc',
-                'resourcequota',
-                'services'
-            ]
-
-            # nodes and pvs are not namespaced, must pull separately
-            self.add_cmd_output([
-                "{} get -o json nodes".format(kube_cmd),
-                "{} get -o json pv".format(kube_cmd)
-            ])
-            if self.get_option('all'):
-                for n in knsps:
-                    knsp = '--namespace=%s' % n
-                    k_cmd = '%s %s %s' % (kube_cmd, kube_get_cmd, knsp)
-
-                    self.add_cmd_output('%s events' % k_cmd)
+        kube_cmd = "kubectl "
+        kubeconfig_path = ""
+        if self.get_option('kubeconfig'):
+            kubeconfig_path = self.get_option('kubeconfig')
+        else:
+            kubeconfig_path = expanduser("~/.kube/config")
+        if kubeconfig_path != "" and path.exists(kubeconfig_path):
+            kube_cmd += "--kubeconfig=" + kubeconfig_path
+
+        kube_get_cmd = "get -o json "
+        for subcmd in ['version', 'config view']:
+            self.add_cmd_output('%s %s' % (kube_cmd, subcmd))
+
+        # get all namespaces in use
+        kn = self.get_command_output('%s get namespaces' % kube_cmd)
+        knsps = [n.split()[0] for n in kn['output'].splitlines()[1:] if n]
+
+        resources = [
+            'limitrange',
+            'pods',
+            'pvc',
+            'rc',
+            'resourcequota',
+            'services'
+        ]
 
-                    for res in resources:
-                        self.add_cmd_output('%s %s' % (k_cmd, res))
-
-                    if self.get_option('describe'):
-                        # need to drop json formatting for this
-                        k_cmd = '%s get %s' % (kube_cmd, knsp)
-                        for res in resources:
-                            r = self.get_command_output(
-                                '%s %s' % (k_cmd, res))
-                            if r['status'] == 0:
-                                k_list = [k.split()[0] for k in
-                                          r['output'].splitlines()[1:]]
-                                for k in k_list:
-                                    k_cmd = '%s %s' % (kube_cmd, knsp)
-                                    self.add_cmd_output(
-                                        '%s describe %s %s' % (k_cmd, res, k))
-
-                if self.get_option('podlogs'):
+        # nodes and pvs are not namespaced, must pull separately
+        self.add_cmd_output([
+            "{} get -o json nodes".format(kube_cmd),
+            "{} get -o json pv".format(kube_cmd)
+        ])
```

Comment:
Worth moving this command (and it's comment) before `# get all namespaces in use`? Since getting all namespaces logically precedes the `if self.get_option('all'):` command / code below, while getting the jsons is independent?

Reply:
It make sense, @pmoravec . I updated that.